staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2733,3 +2745,18 @@ func (p PreferenceMatcher) GetName() string {
 func (p PreferenceMatcher) GetRevisionName() string {
 	return p.RevisionName
 }
+
+type LiveUpdateFeatures struct {
+	// LiveUpdateCPU holds hotplug configuration for the CPU resource.
+	CPU *LiveUpdateCPU `json:"cpu,omitempty" optional:"true"`
+}
+
+type LiveUpdateCPU struct {
+	// The maximum amount of sockets that can be hot-plugged to the Virtual Machine
+	MaxSockets *uint32 `json:"maxSockets,omitempty" optional:"true"`
+}
+
+type LiveUpdateConfiguration struct {
+	// MaxCpuSockets holds the maximum amount of sockets that can be hotplugged
+	MaxCpuSockets uint32 `json:"maxCpuSockets,omitempty"`
```

Comment:
Once we will expand on this API we will need to be able to specify only one of the others defaults. 

In other words, today the following
```
liveUpdateConfiguration: {}
``` 
means the default maximum amount of sockets is 0. 
In future
```
liveUpdateConfiguration: 
    maxMemory: 512Gi
```
would also mean the default maximum amount of sockets is 0. 

I think we want a pointer here to be able to recognize the user's intention and/or be more explicit.